client/layout/masterbar/checkout.tsx

```diff
@@ -54,12 +54,13 @@ const CheckoutMasterbar = ( {
 	const { responseCart, replaceProductsInCart } = useShoppingCart( cartKey );
 	const [ isModalVisible, setIsModalVisible ] = useState( false );
 
-	const closeAndLeave = () =>
+	const closeAndLeave = ( userHasClearedCart: boolean = false ) =>
```

Comment:
This is relatively minor but I think a better API here would be to use object destructuring so this can be a "named" variable. That way instead of `closeAndLeave(true)` you'd see `closeAndLeave({userHasClearedCart:true})` which is much more readable, especially since the property's effect is not obviously related to closing checkout. In addition, it leaves open the option to pass more named variables in the future if we need to.

Reply:
That makes sense. I'll update.